platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/pax/web/jetty/JettyAuthenticator.java

```diff
@@ -11,80 +11,73 @@
-import javax.servlet.FilterChain;
-import javax.servlet.FilterConfig;
-import javax.servlet.ServletContext;
-import javax.servlet.ServletException;
 import javax.servlet.ServletRequest;
 import javax.servlet.ServletResponse;
 import org.apache.commons.collections.CollectionUtils;
 import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.platform.filter.AuthenticationChallengeException;
+import org.codice.ddf.platform.filter.AuthenticationException;
 import org.codice.ddf.platform.filter.SecurityFilter;
+import org.eclipse.jetty.security.ConstraintSecurityHandler;
+import org.eclipse.jetty.security.IdentityService;
+import org.eclipse.jetty.security.ServerAuthException;
+import org.eclipse.jetty.security.authentication.LoginAuthenticator;
+import org.eclipse.jetty.server.Authentication;
+import org.eclipse.jetty.server.UserIdentity;
 import org.osgi.framework.Bundle;
 import org.osgi.framework.BundleContext;
 import org.osgi.framework.FrameworkUtil;
 import org.osgi.framework.InvalidSyntaxException;
 import org.osgi.framework.ServiceReference;
-import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/**
- * {@link DelegateServletFilter} is meant to detect any {@link SecurityFilter}s and doFilter through
- * those before continuing on its {@link FilterChain}. If no {@link SecurityFilter}s are found, it
- * becomes a pass-through filter.
- */
-public class DelegateServletFilter implements Filter {
-
-  private static final Logger LOGGER = LoggerFactory.getLogger(DelegateServletFilter.class);
+public class JettyAuthenticator extends LoginAuthenticator {
 
+  private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(JettyAuthenticator.class);
   private CopyOnWriteArraySet<String> keysOfInitializedSecurityFilters;
 
-  private FilterConfig filterConfig;
-
-  public DelegateServletFilter() {
+  public JettyAuthenticator() {
+    super();
     keysOfInitializedSecurityFilters = new CopyOnWriteArraySet<>();
-  }
-
-  BundleContext getContext() {
-    final Bundle cxfBundle = FrameworkUtil.getBundle(DelegateServletFilter.class);
-    if (cxfBundle != null) {
-      return cxfBundle.getBundleContext();
-    }
-    return null;
+    _loginService = new DummyLoginService();
+    _identityService = _loginService.getIdentityService();
   }
 
   @Override
-  public void init(FilterConfig filterConfig) throws ServletException {
-    this.filterConfig = filterConfig;
-    // need to (re)initialize every SecurityFilter after the DelegateServletFilter is initialized
+  public void setConfiguration(AuthConfiguration configuration) {
     keysOfInitializedSecurityFilters.clear();
+    if (configuration instanceof ConstraintSecurityHandler) {
+      ((ConstraintSecurityHandler) configuration).setLoginService(_loginService);
+      ((ConstraintSecurityHandler) configuration).setIdentityService(_identityService);
+    }
   }
 
   @Override
-  public void destroy() {
-    // do nothing
+  public String getAuthMethod() {
+    return "DDF";
```

Comment:
❓ Does it make sense to use a common constant for this?

Reply:
Yeah, I can make a constant here in `JettyAuthenticator` that is referenced in `SecurityAuthService`, the one in `JettyAuthenticatedUser` isn't used and can just return null.